islandora.module

```diff
@@ -871,7 +873,7 @@ function islandora_user_access($object_or_datastream, array $permissions, $conte
  * @return bool
  *   TRUE if the user is allowed to access this object, FALSE otherwise.
  */
-function islandora_object_access_callback($perm, $object = NULL) {
+function islandora_object_access_callback($perm, AbstractObject $object = NULL) {
```

Comment:
So this line is still blowing up for me in two instances I tested: 

1) Embargoed object. 
2) Object protected with XACML policy.

Both trigger: 
```
TypeError: Argument 2 passed to islandora_object_access_callback() must be an instance of AbstractObject, string given, called in /var/www/drupal/includes/menu.inc on line 648 in islandora_object_access_callback() (line 876 of /var/www/drupal/sites/all/modules/islandora/islandora/islandora.module).
```